### PR TITLE
[nuvo] Add source menu channel

### PR DESCRIPTION
**Pull request by mlobstein**
Follow up to #12042. In the previous PR, the capability was added for the user to create a custom menu tree that would be displayed in the physical keypads. By using rules, the selection of items in the keypad menu can trigger actions in openHAB. However there is currently no capability to easily trigger these menu items from within the openHAB ui. This PR adds a new channel that is automatically populated with state options representing the keypad menu tree. Selecting one of the options has the same effect as choosing that menu item on the keypad.

### CODE REVIEW

File: bundles/org.openhab.binding.nuvo/src/main/java/org/openhab/binding/nuvo/internal/handler/NuvoHandler.java [hunk 1]
```diff
@@ -1114,6 +1119,20 @@ private void enableNuvonet(boolean showReady) {
                                     + topMenuItems.get(i).getText() + "\"");
                             Thread.sleep(SLEEP_BETWEEN_CMD_MS);
                         }
+
+                        // Build a State options selection that represents this source's custom menu
+                        List<StateOption> sourceMenuStateOptions = new ArrayList<>();
+                        topMenuItems.forEach(topItem -> {
+                            sourceMenuStateOptions.add(new StateOption(topItem.getText(), topItem.getText()));
+                            topItem.getItems().forEach(subItem -> {
```
Comment (clinique) [inline bundles/org.openhab.binding.nuvo/src/main/java/org/openhab/binding/nuvo/internal/handler/NuvoHandler.java:1127]: If they is only one statement, you can avoid using `{....;}`

Comment (mlobstein) [reply bundles/org.openhab.binding.nuvo/src/main/java/org/openhab/binding/nuvo/internal/handler/NuvoHandler.java]: fixed

File: bundles/org.openhab.binding.nuvo/src/main/java/org/openhab/binding/nuvo/internal/NuvoBindingConstants.java [hunk 2]
```diff
@@ -66,6 +66,7 @@ public class NuvoBindingConstants {
     public static final String CHANNEL_BUTTON_PRESS = "button_press";
     public static final String CHANNEL_ART_URL = "art_url";
     public static final String CHANNEL_ALBUM_ART = "album_art";
+    public static final String CHANNEL_SOURCE_MENU = "source_menu";
```
[the remark this answers is not in the document]
Comment (mlobstein) [reply bundles/org.openhab.binding.nuvo/src/main/java/org/openhab/binding/nuvo/internal/NuvoBindingConstants.java:69]: Agree but I was not as informed on the standards when I originally wrote some of these bindings. I took cues from other older bindings like dscalarm that use underscores. I propose leaving this unchanged. I don't have any other enhancement planned for this binding.

File: bundles/org.openhab.binding.nuvo/src/main/resources/OH-INF/thing/channels.xml [hunk 3]
```diff
@@ -536,4 +541,10 @@
 		<state readOnly="true"/>
 	</channel-type>
 
+	<channel-type id="sourceMenu">
```
Comment (jlaur) [inline bundles/org.openhab.binding.nuvo/src/main/resources/OH-INF/thing/channels.xml:544]: To follow the convention of the existing channel types in this binding, the id should be:
```suggestion
	<channel-type id="source_menu">
```

Or to follow the openHAB naming convention:
```suggestion
	<channel-type id="source-menu">
```

**Review by jlaur — CHANGES_REQUESTED** [review]
Thanks! Only one minor comment.

File: bundles/org.openhab.binding.nuvo/src/main/resources/OH-INF/thing/channels.xml (hunk 3 above)
Comment (mlobstein) [reply bundles/org.openhab.binding.nuvo/src/main/resources/OH-INF/thing/channels.xml]: Thanks, changed to source_menu for all instances.

**Review by jlaur — APPROVED** [review]
LGTM

**Pull request merged**